### ISSUE TITLE
False positive: Undefined variable '__annotations__' in class body

### ISSUE THREAD

**Issue by dnpetrov** [pylint-dev/pylint#9611]
### Bug description

```python
class Test:
    a: int
    b: float
    c: str

    __slots__ = tuple(__annotations__)
```

### Command used

```shell
pylint ./test.py
```

### Pylint output

```python
************* Module test
test.py:6:22: E0602: Undefined variable '__annotations__' (undefined-variable)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.3
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```

### OS / Environment

Ubuntu 22.04

**Comment by mbyrnepr2** [comment]
Thank you for the report but I will close it since there already exists another issue which captures this scenario: https://github.com/pylint-dev/pylint/issues/7126#issuecomment-1327742196
